design/baremetal-operator/uefi-http-boot.md

```diff
@@ -0,0 +1,102 @@
+# Add UEFI HTTP boot hardware driver
+
+## Status
+
+Implementable
+
+## Summary
+
+This design proposes to add a new hardware driver `redfish-http-boot` to
+support the UEFI HTTP boot interface using the Redfish API.
+
+## Motivation
+
+HTTP boot is available since at least 9 years, it was standardized in UEFI
+version 2.5, and it's considered as a natural valid replacement for
+PXE boot.
+It addresses PXE issues for security, relying on HTTPs and using a
+reliable TCP connection, and scalability and performance, for example with
+the possibility of using HTTP load balancers.
+In general HTTP boot grants a faster, more reliable and more secure way to
+obtain operating system images and EFI executables during a network boot
+process.
+
+### Goals
+
+Support UEFI HTTP boot interface using Redfish API.
+
+### Non-Goals
+
+- HTTP boot is a UEFI only feature, so support for Legacy (BIOS) can't and
+won't be taken into consideration.
+- Any non-Redfish support for HTTP boot.
```

Comment:
We need to make it clear that the current approach relies on booting ISOs, not a kernel/ramdisk pair.